rasa/shared/core/domain.py

```diff
@@ -1954,6 +1956,36 @@ def _validate_forms(forms: Union[Dict, List], domain_slots: Dict[Text, Any]) ->
                     f"mapped in domain slots."
                 )
 
+        all_form_slots = [
+            required_slots[REQUIRED_SLOTS_KEY] for required_slots in forms.values()
+        ]
+        all_required_slots = set(chain.from_iterable(all_form_slots))
+
+        for slot, properties in domain_slots.items():
+            for mapping in properties["mappings"]:
+
+                if "conditions" in mapping:
+                    for condition in mapping["conditions"]:
+                        if (
+                            condition["active_loop"] == form_name
+                            and slot not in form_slots
+                        ):
+                            raise InvalidDomain(
+                                f"Slot '{slot}' has a mapping condition for form "
+                                f"'{form_name}', but it's not present in form "
+                                f"'{REQUIRED_SLOTS_KEY}'."
```

Comment:
`REQUIRED_SLOTS_KEY` seems to be the wrong thing here - shouldn't it be the name of the form?

Reply:
Oh both should be there, I wanted to say the form_name's `required_slots`.